single-pool/js/packages/client/src/index.ts

```diff
@@ -0,0 +1,21 @@
+import { getBase58EncodedAddressCodec } from '@solana/web3.js';
+
+import { PoolAddress, VoteAccountAddress } from './addresses';
+
+export * from './addresses';
+export * from './instructions';
+export * from './transactions';
+
+export async function getVoteAccountAddressForPool(
+  rpc: any, // XXX not exported: Rpc<GetAccountInfoApi>,
```

Comment:
Works!

```ts
import { GetAccountInfoApi } from '@solana/rpc-core/dist/types/rpc-methods/getAccountInfo';
import { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
```

Reply:
oh i see. is this a normal thing to do? i would never have thought to use this weird path and im not sure how users would know to

Reply:
moot point anyway

```
src/index.ts:2:21 - error TS2307: Cannot find module '@solana/rpc-transport/dist/types/json-rpc-types' or its corresponding type declarations.

2 import { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in src/index.ts:2
```

i did end up finding it in `@solana/web3.js/dist/types/rpc` but:

```
src/index.ts:2:10 - error TS2459: Module '"@solana/web3.js/dist/types/rpc"' declares 'Rpc' locally, but it is not exported.

2 import { Rpc } from '@solana/web3.js/dist/types/rpc';
           ~~~

  ../../node_modules/.pnpm/@solana+web3.js@2.0.0-experimental.21e994f_node-fetch@2.7.0_ws@8.14.2/node_modules/@solana/web3.js/dist/types/rpc.d.ts:3:15
    3 import type { Rpc, RpcSubscriptions } from '@solana/rpc-transport/dist/types/json-rpc-types';
                    ~~~
    'Rpc' is declared here.


Found 1 error in src/index.ts:2
```

is this really sensible? cant we just export it normally?

Reply:
FWIW I've been using `type SolanaRpc = ReturnType<typeof createSolanaRpc>;` to get the `Rpc` type with all Solana RPC methods. I didn't know you could import from the dist path either.

Reply:
ok i think im just leaving this as `any` i cant stand this pr anymore

```
src/index.ts:15:36 - error TS2339: Property 'data' does not exist on type 'Readonly<{ context: Readonly<{ slot: bigint; }>; value: Readonly<{ executable: boolean; lamports: LamportsUnsafeBeyond2Pow53Minus1; owner: Base58EncodedAddress; rentEpoch: bigint; }> | null; }> & Readonly<...>'.

15   if (!(poolAccount && poolAccount.data[0] === 1)) {
```